ironfish/src/wallet/wallet.ts

```diff
@@ -428,6 +429,30 @@ export class Wallet {
     }
   }
 
+  private async upsertAssetsFromDecryptedNotes(
+    account: Account,
+    decryptedNotes: DecryptedNote[],
+    tx?: IDatabaseTransaction,
+  ): Promise<void> {
+    for (const { serializedNote } of decryptedNotes) {
+      const note = new Note(serializedNote)
+      const asset = await this.walletDb.getAsset(account, note.assetId(), tx)
+
+      if (!asset) {
+        const chainAsset = await this.chain.getAssetById(note.assetId())
+        Assert.isNotNull(chainAsset, 'Asset must be non-null in the chain')
```

Comment:
I'm gonna write a future note here because we had some out of github conversation on this.

If this doesn't exist in the chain, it can crash. It's unclear how often this can happen so we're gonna ignore this for now. If it happens more, we'll need to figure out a way to save a pending asset, and then add the details later.